.github/actions/configure-environment/action.yml

```diff
@@ -13,11 +13,38 @@ runs:
         /#.*arch=/ && $0 ~ "arch="arch { print }
         ' environment.yml > ci-environment.yml
 
-    - uses: conda-incubator/setup-miniconda@v2
+    - name: Setup Mambaforge
+      uses: conda-incubator/setup-miniconda@v2
       with:
+        miniforge-variant: Mambaforge
+        miniforge-version: latest
+        # Do not specify environment file - see Cache step below
         activate-environment: yardl
-        environment-file: ci-environment.yml
+        use-mamba: true
+
+    - name: Get Date
+      id: get-date
+      shell: bash
+      run: echo "DATE=$(date -u +'%Y%m%d')" >> $GITHUB_ENV
+
+    - name: Cache Conda Env
+      id: cache-conda
+      uses: actions/cache@v3
+      with:
+        path: ${{ env.CONDA }}/envs
+        key:
+            conda-${{ runner.os }}--${{ runner.arch }}--${{
+            hashFiles('ci-environment.yml') }}-${{
```

Comment:
Do we want to throw the conda version in there too?

Reply:
I don't think it's necessary. The cache is invalidated every 24 hours, so within that time period, if the conda version changes *and* somehow the resolved set of packages for our `ci-environment.yml` changes we can easily invalidate the cache on this page: https://github.com/microsoft/yardl/actions/caches.

Reply:
Makes sense.